sdk/storage/azure-storage-file-datalake/tests/test_datalake_service_client.py

```diff
@@ -411,3 +412,47 @@ def test_azure_named_key_credential_access(self, **kwargs):
 
         # Assert
         assert props is not None
+
+    @DataLakePreparer()
+    @recorded_by_proxy
+    def test_datalake_clients_properly_close(self, **kwargs):
+        datalake_storage_account_name = kwargs.pop("datalake_storage_account_name")
+        datalake_storage_account_key = kwargs.pop("datalake_storage_account_key")
+
+        self._setup(datalake_storage_account_name, datalake_storage_account_key)
+        file_system_client = self.dsc.get_file_system_client(file_system='testfs')
+        dir_client = self.dsc.get_directory_client(file_system='testfs', directory='testdir')
+        file_client = dir_client.get_file_client(file='testfile')
+
+        # Mocks
+        self.dsc._blob_service_client.close = MagicMock()
+        self.dsc._client.__exit__ = MagicMock()
+        file_system_client._client.__exit__ = MagicMock()
+        file_system_client._datalake_client_for_blob_operation.close = MagicMock()
+        dir_client._client.__exit__ = MagicMock()
+        dir_client._datalake_client_for_blob_operation.close = MagicMock()
+        file_client._client.__exit__ = MagicMock()
+        file_client._datalake_client_for_blob_operation.close = MagicMock()
+
+        # Act
+        with self.dsc as dsc:
+            dsc.create_file_system(file_system='testfs')
```

Comment:
Mocks look good but we probably could get away with not making any service calls, therefore not needing a recording or `DataLakePreparer()` or anything.

Can probably use `pass` inside each `with` if it complains that you need something inside the `with`.